projects/frontend/cicd/extract_shared_rc_increment.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash -e
+
+# Copyright 2021-2023 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+# Extracts the latest rc-version number for the pipeline id that's passed
+# and returns the incremented version number
+if [ $# -eq 0 ]
+  then
+    echo "ERROR: No argument for pipeline id supplied"
+    exit 3
+fi
+
+out=$(npm view @versatiledatakit/shared versions --json | grep $1-rc | cut -d \" -f 2 | tail -1 | awk -F '.' '{print $4}')
```

Comment:
How do we know for certian in what order `npm view` returns the versions? We seem to rely on that to take the last version. But i could not see anything reading `npm help view` .

Reply:
It's true, it's not mentioned anywhere in the docs. However, based on experiments I've done, it does sort them by version and not by string. So it will do something like

```
  "1.1.909410501-rc.0",
  "1.1.909410501-rc.1",
  "1.1.909410501-rc.2",
  "1.1.909410501-rc.11",
  "1.1.909410501-rc.12",
  "1.1.909410501-rc.13",
```

which is what we want, but I haven't seen it do

```
  "1.1.909410501-rc.0",
  "1.1.909410501-rc.1",
  "1.1.909410501-rc.11",
  "1.1.909410501-rc.12",
  "1.1.909410501-rc.13",
  "1.1.909410501-rc.2",
```

or having them in random order. Again, it's not mentioned anywhere, but logically, it should sort them. Otherwise, this whole command is pretty useless.

Reply:
OK